components/selectors/bloom.rs

```diff
@@ -7,10 +7,13 @@
 use fnv::FnvHasher;
 use std::hash::{Hash, Hasher};
 
+// The top 12 bits of the 32-bit hash value are not used by the bloom filter.
```

Comment:
This comment seems incorrect to me. Based on a cursory reading of this PR, it seems that the hashes are 24 bits, leaving 8 bits unused in a 32 bit word.